core/drand_test.go

```diff
@@ -37,228 +32,324 @@ func setFDLimit() {
+		t.Log("[reshare] Start reshare")
+		group, err := dt.RunReshare(t, nil, nodesToKeep, nodesToAdd, newThreshold, timeout, false, false, false)
 		require.NoError(t, err)
 		doneReshare <- group
 	}()
 	time.Sleep(3 * time.Second)
-	fmt.Printf("\n -- Move to Response phase !! -- \n")
-	dt.MoveTime(timeout)
-	// at this point in time, nodes should have gotten all deals and send back
-	// their responses to all nodes
+
+	t.Log("Move to response phase")
+	dt.AdvanceMockClock(t, timeout)
+	// at this point in time, nodes should have gotten all deals and send back their responses to all nodes
 	time.Sleep(getSleepDuration())
-	fmt.Printf("\n -- Move to Justif phase !! -- \n")
-	dt.MoveTime(timeout)
+
+	t.Log("Move to justification phase")
+	dt.AdvanceMockClock(t, timeout)
 	// at this time, all nodes received the responses of each other nodes but
 	// there is one node missing so they expect justifications
 	time.Sleep(getSleepDuration())
-	fmt.Printf("\n -- Move to Finish phase !! -- \n")
-	dt.MoveTime(timeout)
+
+	t.Log("Move to finish phase")
+	dt.AdvanceMockClock(t, timeout)
 	time.Sleep(getSleepDuration())
 	// at this time they received no justification from the missing node so he's
-	// exlucded of the group and the dkg should finish
+	// excluded of the group and the dkg should finish
 	// time.Sleep(10 * time.Second)
+
 	var resharedGroup *key.Group
 	select {
 	case resharedGroup = <-doneReshare:
 	case <-time.After(1 * time.Second):
 		require.True(t, false)
 	}
-	fmt.Println(" RESHARED GROUP:", resharedGroup)
-	target := resharedGroup.TransitionTime
+
+	t.Logf("[reshare] Group: %s", resharedGroup)
+
+	transitionTime := resharedGroup.TransitionTime
 	now := dt.Now().Unix()
+
 	// get rounds from first node in the "old" group - since he's the leader for
 	// the new group, he's alive
-	lastBeacon := dt.TestPublicBeacon(dt.Ids(1, false)[0], false)
+	t.Log("Check Beacon Public on Leader")
+	lastBeacon := dt.CheckPublicBeacon(dt.Ids(1, false)[0], false)
+
 	// move to the transition time period by period - do not skip potential
 	// periods as to emulate the normal time behavior
-	for now < target-1 {
-		dt.MoveTime(beaconPeriod)
-		lastBeacon = dt.TestPublicBeacon(dt.Ids(1, false)[0], false)
+	for now < transitionTime-1 {
+		dt.AdvanceMockClock(t, beaconPeriod)
+
+		t.Log("Check Beacon Public on Leader")
+		lastBeacon = dt.CheckPublicBeacon(dt.Ids(1, false)[0], false)
 		now = dt.Now().Unix()
 	}
+
 	// move to the transition time
-	dt.MoveToTime(resharedGroup.TransitionTime)
+	dt.SetMockClock(t, resharedGroup.TransitionTime)
 	time.Sleep(getSleepDuration())
+
 	// test that all nodes in the new group have generated a new beacon
-	dt.TestBeaconLength(int(lastBeacon.Round+1), true, dt.Ids(newN, true)...)
+	t.Log("Check Beacon Length")
+	dt.CheckBeaconLength(int(lastBeacon.Round+1), true, dt.Ids(newNodes, true)...)
 }
 
-func TestDrandResharePreempt(t *testing.T) {
+// Test ??????????????????????
+func TestRunDKGResharePreempt(t *testing.T) {
```

Comment:
This is where a client can stop the resharing in process and start again (IIRC)

Reply:
Great!